interface/backend/images/models.py

```diff
@@ -1,18 +1,147 @@
+    rescale_slope = models.IntegerField(null=True)
+    rows = models.IntegerField(null=True)
+    columns = models.IntegerField(null=True)
+    pixel_spacing_col = models.FloatField(null=True)
+    pixel_spacing_row = models.FloatField(null=True)
+    path = models.FilePathField(path=settings.DATASOURCE_DIR,
+                                recursive=True,
+                                allow_files=True,
+                                allow_folders=False,
+                                max_length=512)
+
+    class Meta:
+        ordering = ['slice_location']
+
+    def save(self, *args, **kwargs):
+        self._populate_dicom_properties()
+        super().save(*args, **kwargs)
+
+    def _populate_dicom_properties(self):
+        """ Parse DICOM properties from the file and store the relevant ones
+            on the object.
+        """
+        logger = logging.getLogger(__name__)
+        for k, v in self.load_dicom_data_from_disk(self.path)['metadata'].items():
+            # check that the field is a property on the object
+            if not hasattr(self.__class__, k):
+                logger.warning(f"Property '{k}' not a field on ImageFile, discarding value '{v}'.")
+
+            else:
+                self.__setattr__(k, v)
+
+    @classmethod
+    def load_dicom_data_from_disk(cls, filepath, parse_metadata=True, encode_image_data=False):
+        """ Loads the dicom properties that we care about from disk.
+
+            For the values we load, see ImageFile.DICOM_PROPERTIES
+
+            Args:
+                filepath (str): the path to the file (so this can be used if ImageFile is not in database yet)
+                parse_metadata (bool, optional): weather or not to parse any metadata from the file (skip for just img)
+                encode_image_data (bool, optional): Base64 encode image data and return it.
+        """
+        if not os.path.abspath(filepath).startswith(settings.DATASOURCE_DIR):
+            raise PermissionDenied
+
+        dcm_data = dicom.read_file(filepath)
+
+        # parse metadata
+        metadata = cls._parse_metadata(dcm_data, filepath) if parse_metadata else dict()
+
+        # encode image data as base64 if asked for it
+        image = cls._dicom_to_base64(dcm_data) if encode_image_data else None
+
+        return dict(metadata=metadata, image=image)
+
+    @classmethod
+    def _parse_metadata(cls, dcm_data, filepath):
+        """ Parses metadata from dicom file and creates a dictionary
+            which we can use to populate the ImageFile object
+        """
+        logger = logging.getLogger(__name__)
+
+        # only allow a key to be read from a single DICOM property
+        # and warn on attempted overwrite
+        def _update_once(d1, d2):
+            for k, v in d2.items():
+                if k in d1:
+                    logger.warning(f"Tried to set '{k}' more than once, ignoring value '{v}'.")
+                else:
+                    d1[k] = v
```

Comment:
perhaps move this func up a level so since it doesn't depend on anything provided by the parent func.